src/os/bluestore/BitMapAllocator.cc

```diff
@@ -182,26 +182,28 @@ int BitMapAllocator::alloc_extents_cont(
   *count = 0;
   assert(alloc_unit);
   assert(!(alloc_unit % m_block_size));
+  assert(!(want_size % alloc_unit));
   assert(!(max_alloc_size % m_block_size));
 
   int64_t nblks = (want_size + m_block_size - 1) / m_block_size;
   int64_t start_blk = 0;
   int64_t need_blks = nblks;
-  int64_t max_blks = max_alloc_size / m_block_size;
+  int64_t cont_blks = alloc_unit / m_block_size;
 
   ExtentList block_list = ExtentList(extents, m_block_size, max_alloc_size);
 
   while (need_blks > 0) {
     int64_t count = 0;
-    count = m_bit_alloc->alloc_blocks_res(
-      (max_blks && need_blks > max_blks) ? max_blks : need_blks, hint, &start_blk);
+    count = m_bit_alloc->alloc_blocks_res(cont_blks, hint, &start_blk);
```

Comment:
This looks really inefficient.. if I'm reading this right, you're calling into alloc_blocks_res for *every* alloc_unit (min_alloc_size).  Is that right?